sourcegraph/sourcegraph
# Code Insights: Add num samples display option

Closes https://github.com/sourcegraph/sourcegraph/issues/45823
Fixes https://github.com/sourcegraph/sourcegraph/issues/38926
Fixes https://github.com/sourcegraph/sourcegraph/issues/42459

## Context
This PR adds yet another drill down filter field. Sort and Limit number points field, it works actually like max (limit) filter, points count on the chart can't be greater than filter value. 

<img width="997" alt="Screenshot 2023-01-19 at 00 31 50" src="https://user-images.githubusercontent.com/18492575/213349106-d3682863-6288-4021-ac6a-064c8f0cf78c.png">

## Backend problem
- Save default filters mutation doesn't actually save numSamples parameter 
- Default value is zero but schema says that it's nullable field (0 default value is weird even it wasn't nullable) 

## Test plan
- Open the dashboard page with any insights 
- Open drill down filters (filter icon in the card)
- Fill in sort and limit num of points field 
- Check that point numbers on the chart is equal or less than current value



## App preview:

- [Web](https://sg-web-vk-add-num-sample-ui-filter.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
